### PR TITLE
[IMP] Move meeting values preparation out of holidays_validate method

### PR DESCRIPTION
**Pull request by adrienpeiffer**
To add possibility of inheritance

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
OPW 652325

**Comment by sbidoul** [comment]
:+1: I hope this harmless refactoring for extensibility can be merged.

**Comment by feketemihai** [comment]
:+1:

**Comment by max3903** [comment]
:+1:

**Comment by gurneyalex** [comment]
@odony this PR illustrates a typical case of the discussion we had during the last odoo experience. Not having thes _prepare_xxx methods makes it hard for us to extend Odoo.

You mentionned that such changes could be envisionned even in the stable version. Maybe 8.0 is too old, but would this be acceptable in 9? 10?

See https://github.com/odoo/odoo/blob/9.0/addons/hr_holidays/hr_holidays.py#L400
and https://github.com/odoo/odoo/blob/10.0/addons/hr_holidays/models/hr_holidays.py#L470

Thanks :smile_cat:

File: addons/hr_holidays/hr_holidays.py [hunk 1]
```diff
@@ -357,6 +357,24 @@ def holidays_first_validate(self, cr, uid, ids, context=None):
...
+            'user_id': leave.user_id.id,
+            'start': leave.date_from,
+            'stop': leave.date_to,
+            'allday': False,
+            'state': 'open',            # to block that meeting date in the calendar
+            'class': 'confidential'
+        }
+        # Add the partner_id (if exist) as an attendee
+        if leave.user_id and leave.user_id.partner_id:
+                meeting_vals['partner_ids'] = [(4, leave.user_id.partner_id.id)]
```
Comment (pedrobaeza) [inline addons/hr_holidays/hr_holidays.py]: This has one indentation level more than expected

**Comment by adrienpeiffer** [comment]
Well spotted @pedrobaeza !

**Comment by mart-e** [comment]
Closing in favour of #16432

**Pull request closed without merging**